File: backend/app/models/meal_event.py
"""Meal event model."""
from datetime import datetime
from app import db


class MealEvent(db.Model):
    """Represents a meal intake event with carbohydrate information."""
    __tablename__ = "meal_events"

    id = db.Column(db.Integer, primary_key=True)
    patient_id = db.Column(
        db.Integer, db.ForeignKey("patients.id"), nullable=False, index=True
    )
    timestamp = db.Column(db.DateTime, nullable=False, index=True)
    carbs_grams = db.Column(db.Float, nullable=False)
    meal_type = db.Column(
        db.String(20), nullable=True
    )  # breakfast, lunch, dinner, snack
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    __table_args__ = (
        db.Index("ix_meal_patient_time", "patient_id", "timestamp"),
    )

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "patient_id": self.patient_id,
            "timestamp": self.timestamp.isoformat(),
            "carbs_grams": self.carbs_grams,
            "meal_type": self.meal_type,
        }

    def __repr__(self) -> str:
        return f"<MealEvent {self.timestamp}: {self.carbs_grams}g carbs>"

File: backend/DiabetesApi/Models/MealEvent.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DiabetesApi.Models;

[Table("meal_events")]
public class MealEvent
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("patient_id")]
    public int PatientId { get; set; }

    [Required]
    [Column("timestamp")]
    public DateTime Timestamp { get; set; }

    [Column("carbs_grams")]
    public double CarbsGrams { get; set; }

    [Column("meal_type")]
    [MaxLength(20)]
    public string? MealType { get; set; } // breakfast, lunch, dinner, snack

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Patient Patient { get; set; } = null!;
}
